app/src/main/java/org/p2p/wallet/moonpay/repository/sell/MoonpaySellRemoteRepository.kt

```diff
@@ -60,18 +64,36 @@ class MoonpaySellRemoteRepository(
     @Throws(MoonpaySellError::class)
     override suspend fun getUserSellTransactions(
         userAddress: Base58String
-    ): List<MoonpaySellTransaction> = doMoonpayRequest {
+    ): List<SellTransaction> = doMoonpayRequest {
+        val response = moonpayServerSideApi.getUserSellTransactions(userAddress.base58Value)
+        val depositWallets = getDepositWalletsForTransactions(response)
+
         mapper.fromNetwork(
-            response = moonpayServerSideApi.getUserSellTransactions(userAddress.base58Value),
-            transactionOwnerAddress = userAddress
+            response = response,
+            depositWallets = depositWallets,
+            selectedFiat = getSellFiatCurrency(),
+            transactionOwnerAddress = userAddress,
         )
     }
 
+    private suspend fun getDepositWalletsForTransactions(
```

Comment:
for each WaitingForDeposit transaction we fetch deposit wallet via another request